### DISCUSSION TITLE
Restore from cloud

### DISCUSSION THREAD

**Discussion by pholthaus** [abraunegg/onedrive#1297] [Q&A] [answered]
I have a large number of local deletions and modified/corrupted a bunch of files.

Is there a way to undo these changes without a full re-sync?
If I do a dry sync now, the deletions would be applied to the cloud, if I am using `--download-only` I don't have any changes (as my local modifications are newer).

In a way, I'd like to know whether it's possible to make the programme to behave like it had a `--cloud-first` option.

Thanks for any tips

**Comment by abraunegg** [comment, marked as answer]
@pholthaus 
> I have a large number of local deletions and modified/corrupted a bunch of files.
> Is there a way to undo these changes without a full re-sync?

Not really ... Can you restore from local backup? Are you using a local file system (eg ZFS) that can perform snapshots on your data so you can rollback to a particular point in time?

> In a way, I'd like to know whether it's possible to make the programme to behave like it had a --cloud-first option.

The client always performs in a cloud first operation, this is why `--local-first` exists to treat your local data as the source of truth.

My suggestion here is to use `--resync --dry-run` to see if this will do what you are seeking to achieve.

**Reply by pholthaus** [reply]
Thanks for the clarification and suggestion @abraunegg 

I have just done a full resync which in the end took less time than expected.
